src/sagas.js

```diff
@@ -802,6 +808,60 @@ function* removeVm (action) {
   yield stopProgress({ vmId: action.payload.vmId, name: 'remove', result })
 }
 
+function* fetchAllEvents (action) {
+  const { userId } = action.payload
+  const events = yield callExternalAction('events', Api.events, { payload: {} })
+
+  if (events.error) {
+    return
+  }
+
+  const internalEvents = events.event
+    ? events.event
+      .filter((event) =>
+        event.severity === 'error' &&
+        event.user &&
+        event.user.id === userId
+      )
+      .map((event) => Api.eventToInternal({ event }))
+    : []
+  yield put(setUserMessages({ messages: internalEvents }))
+}
+
+function* dismissEvent (action) {
+  const { event } = action.payload
+  if (event.get('id')) {
+    const result = yield callExternalAction('dismissEvent', Api.dismissEvent, action)
+
+    if (result.status === 'complete') {
+      yield fetchAllEvents(action)
+    }
+  } else {
+    yield put(dismissUserMessage({ time: event.get('time') }))
+  }
+}
+
+function* clearEvents (action) {
+  const events = yield callExternalAction('events', Api.events, { payload: {} })
+
+  if (events.error) {
+    return
+  }
+
+  const sagaEvents = events.event
+    ? events.event
+      .filter((event) =>
+        event.severity === 'error' &&
+        event.user &&
+        event.user.id === action.payload.userId
+      ).map((event) => callExternalAction('dismissEvent', Api.dismissEvent, { payload: { eventId: event.id } }))
+    : []
+
+  yield all(sagaEvents)
+
+  yield fetchAllEvents(action)
```

Comment:
Why not just push an empty array instead of fetching all the events again?

````js
   yield put(setUserMessages({ messages: [] }))
````

Reply:
Because as I wrote previously, engine is source of truth and all information must be got from engine even if theoretically there is no information.